graylog2-server/src/main/java/org/graylog2/shared/security/tls/SelfSignedCertificate.java

```diff
@@ -165,38 +166,50 @@ private static KeyPair generateKeyPair(SecureRandom random, int bits) {
         return keypair;
     }
 
-    private static X509Certificate generateCertificate(String fqdn, KeyPair keypair, SecureRandom random) throws Exception {
-        final PrivateKey key = keypair.getPrivate();
-
-        // Prepare the information required for generating an X.509 certificate.
-        final X509CertInfo info = new X509CertInfo();
-        final X500Name owner = new X500Name("CN=" + fqdn);
-
-        info.set(X509CertInfo.VERSION, new CertificateVersion(CertificateVersion.V3));
-        info.set(X509CertInfo.SERIAL_NUMBER, new CertificateSerialNumber(new BigInteger(64, random)));
+    private static SubjectKeyIdentifier createSubjectKeyIdentifier(Key publicKey) throws IOException {
+        ASN1InputStream is = null;
         try {
```

Comment:
I think we can use try-with-resources here to avoid the `finally` block: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html